docs/source/tertiary/whole-genome-regression.rst

```diff
@@ -0,0 +1,352 @@
+- ``values``: Genotype values for this header in this sample block.  If the matrix is sparse, contains only non-zero values.
+- ``header_block``: An ID assigned to the block *x* containing this header.
+- ``sample_block``: An ID assigned to the block *x* containing the group of samples represented on this row.
+- ``position``:  An integer assigned to this header that specifies the correct sort order for the headers in this block.
+- ``mu``: The mean of the genotype calls for this header.
+- ``sig``: The standard deviation of the genotype calls for this header.
+
+Sample block mapping
+--------------------
+
+The sample block mapping consists of key-value pairs, where each key is a sample block ID and each value is a list of
+sample IDs contained in that sample block. The order of these IDs match the order of the ``values`` arrays in the block
+genotype DataFrame.
+
+Example
+=======
+
+.. code-block:: python
+
+    from glow.wgr.linear_model import RidgeReducer, RidgeRegression
+    from glow.wgr.functions import block_variants_and_samples, get_sample_ids
+    import numpy as np
+    from pyspark.sql.functions import col, lit
+
+    variants_per_block = 5
+    sample_block_count = 10
+    sample_ids = get_sample_ids(genotypes)
+    block_df, sample_blocks = block_variants_and_samples(
+        genotypes, sample_ids, variants_per_block, sample_block_count)
+
+------------------------
+Dimensionality reduction
+------------------------
+
+The first step in the fitting procedure is to apply a dimensionality reduction to the block matrix *X* using the
+``RidgeReducer``. This is accomplished by fitting multiple ridge models within each block *x* and producing a new block
+matrix where each column represents the prediction of one ridge model applied within one block. This approach to model
+building is generally referred to as **stacking**. We will call the block genotype matrix we started with the
+**level 0** matrix in the stack *X0*, and the output of the ridge reduction step the **level 1** matrix *X1*. The
+``RidgeReducer`` class is used for this step, which is initialized with a list of ridge regularization values (referred
+to here as alpha). Since ridge models are indexed by these alpha values, the ``RidgeReducer`` will generate one ridge
+model per value of alpha provided, which in turn will produce one column per block in *X0*, so the final dimensions of
+matrix *X1* will be *Nx(LxK)*, where *L* is the number of header blocks in *X0* and *K* is the number of alpha values
+provided to the ``RidgeReducer``. In practice, we can estimate a span of alpha values in a reasonable order of
+magnitude based on guesses at the heritability of the phenotype we are fitting.
+
+Initialization
+==============
+
+When the ``RidgeReducer`` is initialized, it will assign names to the provided alphas and store them in a dictionary
+accessible as ``RidgeReducer.alphas``.
+
+.. code-block:: python
+
+    alphas_reducer = np.logspace(2, 5, 10)
```

Comment:
what do the 2, 5, 10 refer to? Is this variants_per_block etc.?

Reply:
These are arbitrary alpha values. It may be cleaner in the docs to only document the case that alphas are generated automatically.